ui/app/AppLayouts/Chat/ChatLayout.qml

```diff
@@ -35,12 +36,89 @@ StackLayout {
 
     property var sectionItemModel
+
+    readonly property var joinedMembers: SortFilterProxyModel {
+        sourceModel: sectionItemModel.allMembers
+
+        filters: AnyOf {
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.Accepted
+            }
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.KickedPending
+            }
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.BannedPending
+            }
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.AwaitingAddress
+            }
+        }
+    }
+    readonly property var bannedMembers: SortFilterProxyModel {
+        sourceModel: sectionItemModel.allMembers
+
+        filters: AnyOf {
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.Banned
+            }
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.UnbannedPending
+            }
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.BannedWithAllMessagesDelete
+            }
+        }
+    }
+    readonly property var pendingMemberRequests: SortFilterProxyModel {
+        sourceModel: sectionItemModel.allMembers
+
+        filters: AnyOf {
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.Pending
+            }
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.AcceptedPending
+            }
+            ValueFilter {
+                roleName: "membershipRequestState"
+                value: Constants.CommunityMembershipRequestState.RejectedPending
+            }
+        }
+    }
+    readonly property var declinedMemberRequests: SortFilterProxyModel {
+        sourceModel: sectionItemModel.allMembers
+
+        filters: ValueFilter {
+            roleName: "membershipRequestState"
+            value: Constants.CommunityMembershipRequestState.Rejected
+        }
+    }
+
+    readonly property var sectionItemWithMembers: {
+        let sectionItemCopy = sectionItemModel
+        sectionItemCopy.members = joinedMembers
+        sectionItemCopy.bannedMembers = bannedMembers
+        sectionItemCopy.pendingMemberRequests = pendingMemberRequests
+        sectionItemCopy.declinedMemberRequests = declinedMemberRequests
+        return sectionItemCopy
+    }
```

Comment:
This little trick makes it so that I don,t have to modify every place we used members. I basically put the right model inside the right property name.

Let me know if there is a cleaner way to do it with some sort of SFPM.

Reply:
I don't think there's an easy way; that said, I think this is overkill. For most common properties, I think you can continue using `sectionItemModel` and only where you need a specific member model, you should pass one of the filtered SFPM models instead of the `members`